src/main/java/com/teragrep/pth_06/planner/walker/conditions/NonStreamingCondition.java

```diff
@@ -0,0 +1,128 @@
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner.walker.conditions;
+
+import com.teragrep.pth_06.config.ConditionConfig;
+import org.jooq.Condition;
+import org.jooq.impl.DSL;
+
+import java.util.Objects;
+
+public final class NonStreamingCondition implements QueryCondition {
+
+    private final String tag;
+    private final String value;
+    private final String operation;
+    private final ConditionConfig config;
+
+    public NonStreamingCondition(String tag, String value, String operation, ConditionConfig config) {
+        this.tag = tag;
+        this.value = value;
+        this.operation = operation;
+        this.config = config;
+    }
+
+    public Condition condition() {
+        final Condition condition;
+        switch (tag.toLowerCase()) {
+            case "index":
+                final QueryCondition index = new IndexCondition(value, operation, false);
+                condition = index.condition();
+                break;
+            case "sourcetype":
+                final QueryCondition sourceType = new SourceTypeCondition(value, operation, false);
+                condition = sourceType.condition();
+                break;
+            case "host":
+                final QueryCondition host = new HostCondition(value, operation, false);
+                condition = host.condition();
+                break;
+            case "earliest":
+            case "index_earliest":
+                final QueryCondition earliest = new EarliestCondition(value);
+                condition = earliest.condition();
+                break;
+            case "latest":
+            case "index_latest":
+                final QueryCondition latest = new LatestCondition(value);
+                condition = latest.condition();
+                break;
+            case "indexstatement":
+                if ("EQUALS".equals(operation) && config.bloomEnabled()) {
+                    final QueryCondition indexStatement = new IndexStatementCondition(value, config);
```

Comment:
couldn't this check be in IndexStatementCondition, it would clean up this part of the code. It already checks if bloom is not enabled and returns noCondition.

Reply:
moved the operation and bloom enabled logic into `IndexStatementCondition` and added tests.